onebusaway-android/src/main/java/org/onebusaway/android/travelbehavior/receiver/TransitionBroadcastReceiver.java

```diff
@@ -0,0 +1,262 @@
+                TravelBehaviorConstants.FIREBASE_ACTIVITY_TRANSITION_FOLDER);
+        long riPrefix = PreferenceUtils.getLong(TravelBehaviorConstants.RECORD_ID, 0);
+        mRecordId = riPrefix++ + "-" + UUID.randomUUID().toString();
+        PreferenceUtils.saveLong(TravelBehaviorConstants.RECORD_ID, riPrefix);
+        pathBuilder.append(mRecordId);
+
+        DocumentReference document = FirebaseFirestore.getInstance().document(pathBuilder.toString());
+
+        document.set(tbi).addOnCompleteListener(task -> {
+            if (task.isSuccessful()) {
+                Log.d(TAG, "Activity transition document added with ID " + document.getId());
+            } else {
+                Log.d(TAG, "Activity transition document failed to be added: " +
+                        task.getException().getMessage());
+                task.getException().printStackTrace();
+            }
+        });
+    }
+
+    private void startSaveArrivalAndDepartureWorker() {
+        Data myData = new Data.Builder()
+                .putString(TravelBehaviorConstants.USER_ID, mUid)
+                .putString(TravelBehaviorConstants.RECORD_ID, mRecordId)
+                .build();
+
+        OneTimeWorkRequest workRequest = new OneTimeWorkRequest.
+                Builder(ArrivalsAndDeparturesDataReaderWorker.class)
+                .setInputData(myData)
+                .build();
+        WorkManager.getInstance().enqueue(workRequest);
+    }
+
+    private void startSaveTripPlansWorker() {
+        Data myData = new Data.Builder()
+                .putString(TravelBehaviorConstants.USER_ID, mUid)
+                .putString(TravelBehaviorConstants.RECORD_ID, mRecordId)
+                .build();
+
+        OneTimeWorkRequest workRequest = new OneTimeWorkRequest.Builder(TripPlanDataReaderWorker.class)
+                .setInputData(myData)
+                .build();
+        WorkManager.getInstance().enqueue(workRequest);
+    }
+
+    private void startSaveDestinationRemindersWorker() {
+        Data myData = new Data.Builder()
+                .putString(TravelBehaviorConstants.USER_ID, mUid)
+                .putString(TravelBehaviorConstants.RECORD_ID, mRecordId)
+                .build();
+
+        OneTimeWorkRequest workRequest = new OneTimeWorkRequest.Builder(DestinationReminderReaderWorker.class)
+                .setInputData(myData)
+                .build();
+        WorkManager.getInstance().enqueue(workRequest);
+    }
+
+    private void requestActivityRecognition() {
+        ActivityRecognitionClient client = ActivityRecognition.getClient(mContext);
+        Intent intent = new Intent(mContext, RecognitionBroadcastReceiver.class);
+        intent.putExtra(TravelBehaviorConstants.RECORD_ID, mRecordId);
+
+        PendingIntent pi = PendingIntent.getBroadcast(mContext, 100, intent, PendingIntent.FLAG_ONE_SHOT);
+        client.requestActivityUpdates(TimeUnit.SECONDS.toMillis(10), pi);
```

Comment:
@cagryInside I also changed this conversion to use TimeUnit - could you please double-check?